README.md

```diff
@@ -1 +1,401 @@
+Method: PUT
+Request Body:
+```json
+{
+  "restaurant": ""
+}
+```
+
+19. Retrieve Chef information
+Description: Retrieves the profile of a chef Chef information based on their ID.
+Endpoint: /chef/me
+Method: GET
+Parameters:
+id (required): The ID of the chef to retrieve.
+Request Body:
+```json
+{
+
+}
+```
+
+20. Retrieve Chef Dishes by chef
+Description: Retrieves a list of dishes offered by a certain chef.
+Endpoint: /chefs/:id/dishes
+Method: GET
+Parameters:
+id (required): The ID of the chef to retrieve dishes from.
+Request Body:
+```json
+{
+  
+}
+```
+
+21. Create Dish
+Description: Allows a chef to create a new dish and add it to their culinary offerings.
+Endpoint: /chef/dish
+Method: POST
+Request Body:
+```json
+{
+  
+    "name": "",
+    "description": "",
+    "price": "",
+    "rating": ""
+
+}
+```
+21. Get All Dishes
+Description: Retrieves all dishes offered by the chef.
+Endpoint: /chef/dish
+Method: GET
+
+22. Get Dish By Id
+Description: Retrieves a specific dish by its ID.
+Endpoint: /chef/dish/:id
+Method: GET
+Parameters:
+id (required): The ID of the dish to retrieve.
+
+23. Delete Dish By Id
+Description: Deletes a specific dish by its ID.
+Endpoint: /chef/dish/:id
+Method: DELETE
+Parameters:
+id (required): The ID of the dish to delete.
+
+24. Create Order
+Description: Creates a new order for the authenticated client.
+Endpoint: /orders
+Method: POST
+Request Body:
+```json
+{
+  
+}
+```
+25. Client Cancel Order
+Description: Allows a client to cancel an order they previously placed.
+Endpoint: /orders/cancel-order/:id
+Method: DELETE
+Parameters:
+id (required): The ID of the order to cancel.
+
+26. Get Chef Orders
+Description: Retrieves a list of orders received by a specific chef.
+Endpoint: /orders/chef-orders
+Method: GET
+
+28. Chef Update Order Status
+Description: Allows a chef to update the status of an order they are handling.
+Endpoint: /orders/chef-update-order-status
+Method: PUT
+Request Body:
+```json
+{
+  "status": ""
+}
+```
+
+
+
+## Examples
+
+Here are some examples of how you can interact with the backend API using various endpoints:
+
+1. Example: Creating a new normal user account
+
+```bash
+curl -X POST http://localhost:3000/auth/signup -H "Content-Type: application/json" -d '{
+  "firstname": "John",
+  "lastname": "Doe",
+  "email": "john.doe@example.com",
+  "password": "secure_password",
+  "role": "chef"
+}'
+```
+
+2. Example: Creating a new client profile
+
+```bash
+curl -X POST http://localhost:3000/clients/ -H "Authorization: Bearer <access_token>" -H "Content-Type: application/json" -d '{
+  "address": "123 Main Street",
+  "contactNumber": "555-123-4567"
+}'
+```
+
+3. Example : Retrieving the client's order history
+
+```bash
+curl -X POST http://localhost:3000/orders -H "Authorization: Bearer <access_token>" -H "Content-Type: application/json" -d '{
+  "items": [
+    { "dishId": "<dish_id>", "quantity": 2 },
+    { "dishId": "<dish_id>", "quantity": 1 }
+  ]
```

Comment:
this is not a request body why is it here? lets make this example creating an order